common/src/main/java/com/wynntils/models/containers/LootChestModel.java

```diff
@@ -36,6 +39,8 @@ public final class LootChestModel extends Model {
     private final Storage<Integer> openedChestCount = new Storage<>(0);
     private final Storage<Integer> dryCount = new Storage<>(0);
     private final Storage<Integer> dryBoxes = new Storage<>(0);
+    private Storage<Integer> dryEmeralds = new Storage<>(0);
+    private Storage<EnumMap<GearTier, Integer>> dryItemTiers = new Storage<>(new EnumMap<>(GearTier.class));
```

Comment:
Without reviewing any further, this is wrong. EnumMaps are unsupported by storage, we only support HashMap, right @magicus? Also, can we just use the Map interface here?

Reply:
(also this should be made more clear)

Reply:
Yes, we should use `Map` in the left hand side. I am actually unsure what happens if we use an EnumMap... I think that depends on how Gson handles it. But HashMap sounds like a safer, and better, bet.

Reply:
Well it does seem to work for me:
```json
      "dryItemTiers": {
        "Normal": 5,
        "Unique": 9,
        "Rare": 4,
        "Legendary": 2
      },
```

Reply:
I looked at the source code of EnumMap and it looks like it's just an optimized variant of Map specifically for Enums. Basically, it does not calculate any hashes but uses the `Enum.ordinal()` integer number as the key.

Reply:
Now that I have slept on it, this will do just fine. 

I remember what the issue was.

When reading from json, GSON will use the type declared (in this case EnumMap). So we want to use a suitable type.

However, when writing to json, Gson will use the type's iterator order. And that is just random for HashMap, so we get large arbitrarily discrepancys each time we save the json file.

That was why I went through the source code and replaced all HashMap and HashSet with TreeMap and TreeSet in configs.

I just checked EnumMap, as can be expected, it is maintained and iterated in enum ordinal order, so this is fine.

However, I think there have crept in other HashMaps in Storage and Config. We should probably add some check for that, if possible...